lightclient/src/chain_config.rs

```diff
@@ -0,0 +1,71 @@
+// Copyright 2019-2023 Parity Technologies (UK) Ltd.
+// This file is dual-licensed as Apache-2.0 or GPL-3.0.
+// see LICENSE for license details.
+
+use serde_json::Value;
+use std::borrow::Cow;
+
+/// Something went wrong building chain config.
+#[non_exhaustive]
+#[derive(thiserror::Error, Debug)]
+pub enum ChainConfigError {
+    /// The provided chain spec is the wrong shape.
+    #[error("Invalid chain spec format")]
+    InvalidSpecFormat,
+}
+
+/// Configuration to connect to a chain.
+pub struct ChainConfig<'a> {
```

Comment:
This can grow to have more options like about max json RPC subs etc if we want to expose them. A user can then either build this config and configure it, or pass a chain spec `&str` directly, which will convert into this easily.